Incorporate feedback about actions and functions (#572)
cap-js/docs: java/application-services.md

}
```

> The `cds-maven-plugin` generates event-specific event context interfaces for the action or function, based on its CDS model definition. These event context interfaces give direct access to the parameters and the return value of the action or function.

Reviewer: ```suggestion
The `cds-maven-plugin` generates event-specific event context interfaces for the action or function, based on its CDS model definition. These event context interfaces provide direct access to the parameters and the return value of the action or function.
```

> The `cds-maven-plugin` generates event-specific event context interfaces for the action or function, based on its CDS model definition. These event context interfaces provide direct access to the parameters and the return value of the action or function.

Reviewer 2: ```suggestion
The `cds-maven-plugin` generates event context interfaces for the action or function, based on its CDS model definition. These event context interfaces provide direct access to the parameters and the return value of the action or function.
```

The `cds-maven-plugin` generates event-specific event context interfaces for the action or function, based on its CDS model definition. These event context interfaces provide direct access to the parameters and the return value of the action or function.
For bound actions or functions the event context interface provides a [CqnSelect](query-api#select) statement, which targets the entity the action or function was triggered on.

> Action-specific event context, generated by the CAP Java SDK Maven Plugin:

Reviewer: but the context is generated for an event here, isn't it?

Author: lets keep the suggestion from Marc.

public class CatalogServiceHandler implements EventHandler {

    @On(event = "review", entity = Books_.CDS_NAME)

>     public void reviewAction(EventContext context) {

Author: ah thanks, will fix it.

>     public void reviewAction(EventContext context) {

Reviewer 2: But I didn't mean to adapt this part. Using the generic context when implementing a concrete action shouldn't be done. What I was referring to was the example below, where the event context is used to trigger the action. Using the typed context interface there is no longer useful -> should use typed Service interface instead. If we want to still describe how to trigger the action via context, then we should give an example using a generic context there.

However examples for implementing an action usually should always use the typed event context interface. As actions are usually not implemented generically.

>     public void reviewAction(EventContext context) {

Author: ok, I will revert my last change and modify the trigger action scenario with a generic event context.

  ...
```

> Alternatively, the event context can be used to trigger the action or function. This approach is useful for the generic use case, where typed interfaces don't work. The event context needs to be filled with the parameter values and emitted on the service:

Author: done, replaced specific context with generic EventContext interface.

  ...
```

Alternatively, the event context can be used to trigger the action or function. This approach is useful for generic use cases, where typed interfaces are not available. The event context needs to be filled with the parameter values and emitted on the service:

```java
    EventContext context = EventContext.create("review", Books_.CDS_NAME);
    context.put("cqn", Select.from(BOOKS).byId("myBookId"));

>     context.put("rating", review.getRating());

Reviewer 2: generic event context will not have `getResult()` method used below.

Author: fixed.

  ...
```

> Alternatively the event context can be used to trigger the action or function. This approach is really usefull for a generic use case, where typed interfaces don't work. Therefore the event context needs to be filled with the parameter values and emitted on the service:

Reviewer: ```suggestion
Alternatively, the event context can be used to trigger the action or function. This approach is useful for the generic use case, where typed interfaces don't work. The event context needs to be filled with the parameter values and emitted on the service:
```
